cmd/config-current.go

```diff
@@ -373,7 +378,8 @@ func loadConfig() error {
 	if !globalIsEnvDomainName {
 		globalDomainName = globalServerConfig.Domain
 	}
-	if !globalIsStorageClass {
+	// Fetch storage class from config only if it is erasure coding setup, else ignore
+	if !globalIsStorageClass && globalIsXL {
 		globalStandardStorageClass, globalRRStorageClass = globalServerConfig.GetStorageClass()
```

Comment:
GetStorageClass() should return an error and must be returned during `loadConfig()`